test/acceptance/features/bindAppToService.feature

```diff
@@ -804,3 +804,103 @@ Feature: Bind an application to a service
+                name: binding-request-backend-ann-sb
+            spec:
+                services:
+                -   group: stable.example.com
+                    version: v1
+                    kind: Backend
+                    name: backend-demo-2
+            """
+        Then jq ".status.conditions[] | select(.type=="CollectionReady").status" of Service Binding "binding-request-backend-ann-sb" should be changed to "True"
+        And jq ".status.conditions[] | select(.type=="InjectionReady").status" of Service Binding "binding-request-backend-ann-sb" should be changed to "False"
+        And Secret "binding-request-backend-ann-sb" contains "BACKEND_HOST" key with value "example.com"
+        And Secret "binding-request-backend-ann-sb" does not contain "BACKEND_READY"
+        # Backend metadata.annotations for service binding is updated
+        When The Custom Resource is present
+            """
+            apiVersion: "stable.example.com/v1"
+            kind: Backend
+            metadata:
+                name: backend-demo-2
+                annotations:
+                    service.binding/host: path={.spec.host}
+                    service.binding/ready: path={.status.ready}
+            spec:
+                host: example.com
+            status:
+                ready: true
+            """
+        Then Secret "binding-request-backend-ann-sb" contains "BACKEND_READY" key with value "true"
+        And Secret "binding-request-backend-ann-sb" contains "BACKEND_HOST" key with value "example.com"
+
+
+   @negative
+   Scenario: Backend Service metadata annotations update not specific to service bindings does not get propagated to the binding secret
+        Given OLM Operator "backend" is running
+        * The Custom Resource is present
+            """
+            apiVersion: "stable.example.com/v1"
+            kind: Backend
+            metadata:
+                name: backend-demo-3
+                annotations:
+                    host : 'binding:env:attribute'
+            spec:
+                host: example.com
+            status:
+                ready:true
+            """
+        * Service Binding is applied
+            """
+            apiVersion: operators.coreos.com/v1alpha1
+            kind: ServiceBinding
+            metadata:
+                name: binding-request-backend-ann
+            spec:
+                services:
+                -   group: stable.example.com
+                    version: v1
+                    kind: Backend
+                    name: backend-demo-3
+            """
+        Then jq ".status.conditions[] | select(.type=="CollectionReady").status" of Service Binding "binding-request-backend-ann" should be changed to "True"
+        And jq ".status.conditions[] | select(.type=="InjectionReady").status" of Service Binding "binding-request-backend-ann" should be changed to "False"
+        And Secret "binding-request-backend-ann" does not contain "BACKEND_HOST"
```

Comment:
why do we need this step?

Reply:
In the scenario where this step is used, there we are applying SBR 2 times. Once with a non specific service-binding-annotation and then update to the similar non specific service binding annotation to check that both the times the secret key should not contain anything. This scenario was added for https://github.com/redhat-developer/service-binding-operator/pull/704#discussion_r498896479

Reply:
hence this shows that when annotations are updated which are non specific to binding, then retrigger does not occur. And to check that we check the content of secret again

Reply:
ah, so secret should not contain anything - but the steps says `Secret "binding-request-backend-ann" does not contain "BACKEND_HOST`. Hence, if it should not contain anything, then the step should be

```
Secret "binding-request-backend-ann" is empty
```

The other question is - should secret be created at all? Should the application deployment be modified at all?

Reply:
Currently, if SBR is there then the secret is created even with data 0.